src/NServiceBus.Core/Performance/Metrics/ReceiveDiagnosticsBehavior.cs

```diff
@@ -0,0 +1,46 @@
+namespace NServiceBus.Performance.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.Metrics;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    class ReceiveDiagnosticsBehavior : IBehavior<IIncomingPhysicalMessageContext, IIncomingPhysicalMessageContext>
```

Comment:
Depending on how the pipeline is built this may capture Subscribe/Unsubscribe requests for message-driven pub-sub. The behaviors that handle those are also in the `IIncomingPhysicalMessageContext` so they may run before or after this behavior.

I don't think that's a big problem overall as most endpoints will not see a large proportion of their message load be subscribe/unsubscribe messages, but I wanted to raise it in case someone else sees an issue with this.

Reply:
@mikeminutillo there are two scenarios I can see:

- If the metrics are used for health checks / devops, occasional extra subscribe/unsubscribe messages are not a big deal (they are a message after all)
- If the metrics are used for licensing calculations, it should be done for a long enough period (e.g. 1 week or month) to average out peak hours, and that would smooth out occasional subscriber/unsubscribe messages.